### PR TITLE
Add libunwind

### PR DESCRIPTION
**Pull request by synapticarbors**
Fixes https://github.com/conda-forge/staged-recipes/issues/3319

### CODE REVIEW

File: recipes/libunwind/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  version: {{ version }}
+
+source:
+  url: http://download.savannah.nongnu.org/releases/libunwind/libunwind-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [win or osx]
+
```
Comment (isuruf) [inline recipes/libunwind/meta.yaml]: Add `toolchain` as a build requirement

**Comment by synapticarbors** [comment]
@conda-forge/staged-recipes -- if someone could take a careful look at this recipe since I've primarily only built python packages in the past, I'd appreciate it. I disabled `make check` since it was failing on a single test and couldn't find anything on that particular error in my googling.

File: recipes/libunwind/meta.yaml (hunk 1 above)
Comment (jakirkham) [reply recipes/libunwind/meta.yaml]: note: Done below.

File: recipes/libunwind/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+requirements:
+  build:
+    - toolchain
+
+test:
+  commands:
+    - test -f ${PREFIX}/lib/libunwind.so
```
Comment (jakirkham) [inline recipes/libunwind/meta.yaml]: Is there a header we should check for?

File: recipes/libunwind/build.sh [hunk 3]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+./configure --prefix=$PREFIX
+make
+# make check
```
Comment (jakirkham) [inline recipes/libunwind/build.sh]: What sort of issues were encountered here?

Comment (synapticarbors) [reply recipes/libunwind/build.sh]: It looked like a single test failed (`run-core-dump-unwind`) with all others passing. I couldn't find anything relevant on the libunwind email list pointing to a specific issue.

File: recipes/libunwind/meta.yaml (hunk 2 above)
Comment (synapticarbors) [reply recipes/libunwind/meta.yaml]: Copied over a few additional checks from the `libosxunwind` recipe

File: recipes/libunwind/build.sh (hunk 3 above)
Comment (synapticarbors) [reply recipes/libunwind/build.sh]: Do you feel it necessary that all of the tests pass? The only thing I could find was this thread that had no resolution:

https://lists.nongnu.org/archive/html/libunwind-devel/2015-01/msg00010.html

Comment (jakirkham) [reply recipes/libunwind/build.sh]: If it were me, I'd patch out the broken test and link to that thread in the recipe comments. The motivation being I'd want to be sure we had tested something this low level to the fullest extent possible.

**Comment by jakirkham** [comment]
LGTM. Just need to settle what we are doing with `make check`.

File: recipes/libunwind/build.sh (hunk 3 above)
Comment (synapticarbors) [reply recipes/libunwind/build.sh]: It's not entirely clear to me where to patch out the test. This is mostly due to my unfamiliarity with autotools and all of the generated files. There are `Makefile.in` and `Makefile.am` files both at the top level of the source distribution, but also in the `tests` directory. I'm not 100% sure where the test is actually being run from.

Comment (jakirkham) [reply recipes/libunwind/build.sh]: I think you might be able to get away with just deleting the file `tests/run-coredump-unwind`. At least that would be an easy first thing to try.

Comment (synapticarbors) [reply recipes/libunwind/build.sh]: Ok, I commented out the logic in `tests/run-coredump-unwind` as a patch and that seems to have worked . All other tests pass and I've added a comment above the `patches` entry in the `meta.yaml`.

File: recipes/libunwind/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: http://download.savannah.nongnu.org/releases/libunwind/libunwind-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    # no-op failing test. No clear explanation for failure. Only discussion went unresolved on upstream mailing
+    # list: https://lists.nongnu.org/archive/html/libunwind-devel/2015-01/msg00010.html
+    - run-coredump-unwind.patch
```
Comment (jakirkham) [inline recipes/libunwind/meta.yaml]: xref: https://lists.nongnu.org/archive/html/libunwind-devel/2015-01/msg00010.html

File: recipes/libunwind/build.sh (hunk 3 above)
Comment (jakirkham) [reply recipes/libunwind/build.sh]: SGTM. Might want to bump that thread. Though it is pretty old.

**Comment by jakirkham** [comment]
Anything else from others?

**Comment by scopatz** [comment]
Thanks @synapticarbors!

**Pull request merged**